### PR TITLE
Jenkins 60649 no webhook on first build

### PR DESCRIPTION
**Pull request by mhenschke-atl**
When defined with an SCM, WorkflowJobs store the SCM inside their definition rather than in the SCM list of the job. This PR adds a branch in the webhook trigger's start method to access the correct SCM in workflow jobs.
I haven't included a functional test in the PR, after some effort I decided the difficulty in having the CspScmDefinition correctly use a BitbucketSCM was not worth the effort. You can see the attempt in an earlier commit of this PR.

### CODE REVIEW

File: src/test/java/it/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookTriggerImplTest.java [hunk 1]
```diff
@@ -184,16 +185,20 @@ public void testTriggerPollingSubsequentBuildSignificantChanges() {
...
 
         Queue<PollingResult> pollingResults = new LinkedList<>();
 
-        void addPollingResult(PollingResult result) {
-            pollingResults.add(result);
+        @Override
+        public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, TaskListener listener,
+                             File changelogFile, SCMRevisionState baseline) throws IOException, InterruptedException {
+            //Place a Jenkinsfile in the working directory
+            TestUtils.copyFileToPath("/it/com/atlassian/bitbucket/jenkins/internal/scm/Jenkinsfile",
```
Comment (dkjellin) [inline src/test/java/it/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookTriggerImplTest.java]: hmm, this seems like a leftover from earlier testing?

File: src/test/java/com/atlassian/bitbucket/jenkins/internal/util/TestUtils.java [hunk 2]
```diff
@@ -28,6 +30,18 @@ public static String readFileToString(String relativeFilename) {
         }
     }
 
+    public static void copyFileToPath(String relativeFilename, String path) {
```
Comment (dkjellin) [inline src/test/java/com/atlassian/bitbucket/jenkins/internal/util/TestUtils.java]: this method is only used from the checkout in the `TestSCM` and that usage should probably not be there? So maybe remove this method as well?

File: src/test/resources/it/com/atlassian/bitbucket/jenkins/internal/scm/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+pipeline {
```
Comment (dkjellin) [inline src/test/resources/it/com/atlassian/bitbucket/jenkins/internal/scm/Jenkinsfile]: I think this whole file can be removed?

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketScmHelper.java [hunk 4]
```diff
@@ -40,7 +40,7 @@ public BitbucketRepository getRepository(String projectName, String repositoryNa
         try {
             BitbucketProject project = getProjectByNameOrKey(projectName, clientFactory);
             try {
-                return getRepositoryByNameOrSlug(project.getName(), repositoryName, clientFactory);
+                return getRepositoryByNameOrSlug(project.getKey(), repositoryName, clientFactory);
```
[the remark this answers is not in the document]
Comment (mhenschke-atl) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketScmHelper.java]: Yes that's unrelated. Thanks for catching

File: src/test/java/it/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookTriggerImplTest.java [hunk 5]
```diff
@@ -184,16 +185,20 @@ public void testTriggerPollingSubsequentBuildSignificantChanges() {
                     }
                     return empty();
                 },
-                30000);
+                10000);
```
Comment (mhenschke-atl) [inline src/test/java/it/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookTriggerImplTest.java]: This is also an unintended change. Will set this back

**Pull request merged**